switch continuous integration system to Drone CI

nodiscc opened the pull request:
See commit messages

nodiscc commented on .drone.yml:
```diff
@@ -0,0 +1,147 @@
+    # pull: always # force update of base image even if cached locally
+    commands:
+      # git clone as drone user
+      - git clone --depth=50 --branch=$DRONE_BRANCH $DRONE_GIT_HTTP_URL .
```
`$DRONE_GIT_HTTP_URL` points to https://github.com/shaarli/Shaarli but the HEAD commit is only present on https://github.com/nodiscc/Shaarli

ArthurHoaro replied:
Wouldn't it fail on any PR, where the commit ID points to the submitter's fork and not Shaarli repository itself?

nodiscc replied:
yes

ArthurHoaro commented on .drone.yml:
```diff
@@ -0,0 +1,147 @@
+    commands:
+      - git clone --depth=50 --branch=$DRONE_BRANCH $DRONE_GIT_HTTP_URL .
+      - git checkout $DRONE_COMMIT
+      - composer config --unset platform && composer config platform.php 7.4
```
Good!
Fixing the PHP version as we do isn't a very good practice. Once we've fixed the PHP 8.0 issue, we can probably remove that setting entirely from `composer.json`.

ArthurHoaro commented on .drone.yml:
```diff
@@ -0,0 +1,147 @@
+steps:
+  - name: test-php-8.0
+    image: shaarli/drone-ci-base:8.0
+    # pull: always # force update of base image even if cached locally
```
Why is this commented? If it's not required we should remove it.

nodiscc replied:
ok. something to be aware of though, in case Dockerfiles at https://github.com/shaarli/drone-ci-base are updated

ArthurHoaro reviewed (commented):
Thank you for work on that part! That's great!

Two other things:

  * could you open a separate improvement issue to use the matrix thingy of Drone, copy/pasting config usually leads to out of sync settings issues.
  * how is this doing with Composer/npm cache?

ArthurHoaro commented:
@nodiscc Would you like some help on this? We are almost out of Travis credits.

nodiscc commented:
> how is this doing with Composer/npm cache?

I don't know. Barebones drone doesn't provide cache as far I know. But there is obviously some sort of caching given the build times/logs I am getting - but I didn't configure anything.

At some point I tried to implement [volume cache](http://plugins.drone.io/drillster/drone-volume-cache/) and a `restore_cache` step, but all I was getting was cache misses. I think the CI instance/runner has to explicitly enable such plugins/volume mounting, and I'm pretty sure cloud.drone.io doesn't. Another solution is to use a cache plugin backed by S3 storage (paid, expensive).

> separate improvement issue to use the matrix thingy of Drone

jsonnet https://docs.drone.io/pipeline/kubernetes/examples/language/rust/#test-multiple-versions ? will do

> We are almost out of Travis credits.

:scream: already? damn sorry other things kept me busy. The only remaining problem is `$DRONE_GIT_HTTP_URL` always pointing to the base fork. It's caused by the workaround to prevent drone from cloning the repo as root (`clone:  disable: true` + manually implemented `git clone`). I'll look into it now but any pointers are welcome

nodiscc commented:
Done, rebased. I removed the workaround for `git clone` always being run as root, it appears to work correctly/clone as non-root since `USER drone` was added to https://github.com/shaarli/drone-ci-base. Is there anything else that we should check?

ArthurHoaro commented:
> I don't know. Barebones drone doesn't provide cache as far I know. But there is obviously some sort of caching given the build times/logs I am getting - but I didn't configure anything.

You are right, they probably have something internally, but in any case dependencies + all tests in 18 seconds is not bad at all.

> jsonnet docs.drone.io/pipeline/kubernetes/examples/language/rust/#test-multiple-versions ? will do

Sure it looks great.

> 😱 already? damn sorry other things kept me busy.

No worries, I didn't want to pressure you or anything. 

> Done, rebased. I removed the workaround for git clone always being run as root, it appears to work correctly/clone as non-root since USER drone was added to shaarli/drone-ci-base. Is there anything else that we should check?

Awesome work! 🙌 Thank you.

Everything looks good to me. I think that we can merge this now, and if there are other improvements to add, we can do so later.
